apps/mobile/app/db/init.ts

```diff
@@ -1,20 +1,39 @@
 import * as SQLite from "expo-sqlite";
+import { createSingletonDbProvider } from "../sync/SyncedExpoDB";
+import { cryb64 } from "@vlcn.io/ws-common";
+import {decode, encode} from 'base-64'
+if (!global.btoa) {
+  global.btoa = encode;
+}
+
+if (!global.atob) {
+  global.atob = decode;
+}
 
-export const db = SQLite.openDatabase("test.db");
+export const dbName = "test.db";
+export const db = SQLite.openDatabase(dbName);
+
+// TODO: ideally we can share the schema in a package between client and server
+// this is curently duplicated into /server/schemas/test.sql
+const schema = [
+  `CREATE TABLE IF NOT EXISTS "todo" ("id" PRIMARY KEY, "text", "completed" INTEGER DEFAULT 0);`,
+  `SELECT crsql_as_crr('todo');`
+];
 
 export function initDatabase() {
   db.exec(
-    [
-      {
-        sql: 'CREATE TABLE IF NOT EXISTS "todo" ("id" PRIMARY KEY, "text", "completed" INTEGER DEFAULT 0);',
-        args: [],
-      },
-      {
-        sql: "SELECT crsql_as_crr('todo');",
-        args: [],
-      },
-    ],
+    schema.map((sql) => ({ sql, args: [] })),
     false,
     () => {}
   );
 }
+
+export const dbProvider = createSingletonDbProvider({
+  dbName,
+  db,
+  // TODO: users shouldn't manually deal with any of this.
+  // The browser db wrappers of cr-sqlite support automigration
```

Comment:
cc @alanjhughes - here's something we can follow up on supporting in expo-sqlite